### PR TITLE
Android dump

### PR DESCRIPTION
**Pull request by SachinVin**
All of the stuff from the android side of thing, with changes from https://github.com/citra-emu/citra-android/pull/229 and https://github.com/citra-emu/citra-android/pull/186.

I ended up setting up the build on Github actions too, since it is much nicer to work with than bitrise, and also because ccache doesn't seem to work on bitrise for whatever reason.

Bumped up the versions for other general stuff - gradle, cmake, ndk. Before you ask, `23.1.7779620` is the latest NDK available on Github actions, just using it to speed up the build, we can prolly bump it up if we need to.

Removed the setting for async GPU from the front end for now, as I work on fixing it on desktop with #5845.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6004)

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -117,8 +117,10 @@ endif()
 if (ENABLE_QT)
     add_subdirectory(citra_qt)
 endif()
+
 if (ANDROID)
-    add_subdirectory(android/app/src/main/cpp)
+    include_directories(android/app/src/main)
```
Comment (B3n30) [inline src/CMakeLists.txt:122]: this looks like a way to general way to include files. Can't this be a target_include_directories?

**Review by B3n30 — APPROVED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (SachinVin) [reply src/CMakeLists.txt:122]: You may be right, I'll experiment later

Comment (SachinVin) [reply src/CMakeLists.txt:122]: So, `target_include_directories` requires us to create a library first, do we want to add JNI as "library", seems counter-intuitive?

Comment (B3n30) [reply src/CMakeLists.txt:122]: there is the target main so `target_include_directories(main PRIVATE android/app/src/main)` should do it, right? I don't see any other lib or executable that should use this path as an include dir, or am I missing something?

Comment (SachinVin) [reply src/CMakeLists.txt]: NVM, I misunderstood usage of `target_include_directories`, fixed now.

**Review by bunnei — APPROVED** [review]

**Pull request merged**